### ISSUE TITLE
Gutenberg immediate publishing causes share on publish not to fire.

### ISSUE THREAD

**Issue by cklosowski** [postpromoterpro/post-promoter-pro#313]
When writing a blog post and going immediately to publish (instead of scheduling the blog post) the "Share on publish" features aren't working correctly.

Steps to replicate:
1) Add a new blog post
2) Check the box to share on publish
3) Add a title
4) Add Content
5) Publish immediately

Again, even setting the post to be published in the future 1 minute will allow it to work correctly, it just seems that publishing immediately isn't firing the correct hooks on `transition_post_status`

**Comment by uamv** [comment]
This is probably related – I think I have found that this also fails in the opposite direction. I've noticed on two occasions that if PPP is set to share by default, and I set to prevent sharing immediately prior to publishing a Gutenberg post, it will still share. It seems I need to first save the draft with PPP publishing disabled in order to prevent the share.

**Comment by cklosowski** [comment]
@uamv Yes that seems pretty close. I'm still waiting to hear back from the Gutenberg team on why this regression has been just brushed to the side. It's causing quite a few plugins to fail, not just PPP. There are a few work arounds, but those are not ideal and could stop working at anytime. I'm hoping to help the team see the problem and get the fix actually into Gutenberg.

I'll keep trying to find a workaround that's stable though.

**Comment by cklosowski** [comment]
As a note this is the issue tracking on the Gutenberg project that's causing this:
`/WordPress/gutenberg/issues/12897`

**Comment by cklosowski** [comment]
I played around with this more last night and I can't get something to reliably work. As it's stated in the issue on Gutbenberg there seems to be some inconsistency to the responses I'm getting as well.

I have to punt this to get a Facebook fix out in the meantime #315 out as it's causing issues with Facebook in general.
